testing/snapshot-test.md

```diff
@@ -0,0 +1,20 @@
+# Snapshot test
+
+## Introduction
+Snapshot tests are used to test visual regression of presentational components and for reducers in order to avoid deep comparison boilerplate code
+
+## Dos
+- The visual component output has to be relatively small.  
+- Focus to shallow snapshot comparison.
+- In cases where snapshot output is not readable use enzyme-to-json utility.
+- Create snapshot for every presentational component.  
+
+## Don'ts
+- Don't use snapshots for testing component prop changes. Do assertions on specific DOM nodes directly.
+- Don't use mounted snapshots, better divide components to smaller ones and do shallow snapshots on them directly.
```

Comment:
If we only `shallow` snapshot components, we would lose the CSS rules in the snapshot. I often find these valuable, particularly when it comes to theme colours or breakpoints.

Reply:
I think sometimes we need to use `mount`

Reply:
personally I don't like mounting and snapshotting. If I have to mount I'd normally do a normal enzyme.find() and do a unit test on that.